api/src/api/emailCheckRouter.test.ts

```diff
@@ -0,0 +1,69 @@
+import { emailCheckRouterFactory } from "@api/emailCheckRouter";
+import { UserDataClient } from "@domain/types";
+import { setupExpress } from "@libs/express";
+import { DummyLogWriter } from "@libs/logWriter";
+import { generateUserData } from "@shared/test";
+import { Express } from "express";
+import { StatusCodes } from "http-status-codes";
+import request, { Response } from "supertest";
+
+describe("emailCheckRouter", () => {
+  let app: Express;
+
+  let stubUserDataClient: jest.Mocked<UserDataClient>;
+
+  beforeEach(async () => {
+    jest.resetAllMocks();
+    stubUserDataClient = {
+      get: jest.fn(),
+      put: jest.fn(),
+      findByEmail: jest.fn(),
+      getNeedNewsletterUsers: jest.fn(),
+      getNeedToAddToUserTestingUsers: jest.fn(),
+      getNeedTaxIdEncryptionUsers: jest.fn(),
+      getUsersWithOutdatedVersion: jest.fn(),
+    };
+    app = setupExpress(false);
+    app.use(emailCheckRouterFactory(stubUserDataClient, DummyLogWriter));
+  });
+
+  afterAll(async () => {
+    await new Promise((resolve) => {
+      return setTimeout(resolve, 500);
+    });
+  });
```

Comment:
Why do we need this?

Reply:
This was a copy-paste casualty, but now that you mention it, I'm not sure why this is needed in other files, either? It's in all the `api/src/api/*.test.ts` files, but I commented it out in every test file in the directory and the whole test suite still passed.

My guess is that it was added to avoid some asynchronous timing issues causing tests to fail, but maybe that's a dated assumption and they're safe to remove?

The point is moot for this file, since I'm fine consolidating with `userRouterFactory`, but this could be a good follow-up refactor for the other test files